client/components/Signup.js

```diff
@@ -14,18 +17,41 @@ class Signup extends Component {
                 email: '',
                 password: ''
             },
-            dirty: false
+            dirtyPassword: false,
+            dirtyEmail: false,
+            dirtyUsername: false
         }
         this.handleChange = this.handleChange.bind(this)
         this.handleSubmit = this.handleSubmit.bind(this)
+        this.usernameCheck = this.usernameCheck.bind(this)
+    }
+
+    componentDidMount() {
+        this.props.user.signupError = null
+    }
+
+    usernameCheck() {
+        this.props.usernameAvail({ username: this.state.account.username })
+    }
+
+    validateEmail(email) {
```

Comment:
@AnnaC91  do we need a bind for this class method as well? 

this.validateEmail = this.ValidateEmail.bind(this)  ?